File: merge_strings_alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder merged = new StringBuilder();
        int maxLength = Math.max(word1.length(), word2.length());

        for (int i = 0; i < maxLength; i++) {
            if (i < word1.length()) {
                merged.append(word1.charAt(i));
            }

            if (i < word2.length()) {
                merged.append(word2.charAt(i));
            }
        }

        return merged.toString();        
    }
}

File: merge_strings_alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        std::string merged;
        int maxLength = std::max(word1.length(), word2.length());

        for (int i = 0; i < maxLength; i++) {
            if (i < word1.length()) {
                merged += word1[i];
            }
            
            if (i < word2.length()) {
                merged += word2[i];
            }
        }

        return merged;        
    }
};
